File: 백준/Silver/17087. 숨바꼭질 6/숨바꼭질 6.py

#수빈이 위치 s
n, s =map(int,input().split())
arr = list(map(int,input().split()))
diff = []

def gcd(x,y):
    if x < y:
        x,y = y,x
    if y == 0:
        return x
    return gcd(y, x%y)

for i in arr:
    diff.append(abs(s-i))

if len(diff) == 1:
    print(diff[0])
    exit()
elif len(diff) == 2:
    print(gcd(diff[0],diff[1]))
    exit()
else:
    temp = gcd(diff[0], diff[1])
    for j in range(2,len(diff)):
        temp = gcd(temp, diff[j])
    print(temp)
    exit()


    

File: 백준/Silver/17087. 숨바꼭질 6/숨바꼭질 6.java


import java.util.Arrays;
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int s = sc.nextInt();
        int[] arr = new int[n + 1];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        arr[n] = s;

        Arrays.sort(arr);

        int currentGcd = arr[1] - arr[0];
        for (int i = 2; i < n + 1; i++) {
            currentGcd = gcd(currentGcd, arr[i] - arr[i - 1]);
        }
        System.out.println(currentGcd);

    }

    private static int gcd(int a, int b) {
        if (a < b) {
            int temp = a;
            a = b;
            b = temp;
        }
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

}



